### PR TITLE
fix(upgrade): leak when angular1 destroys element

### LINKED ISSUE #6401: bug(upgrade): leak: ngOnDestroy not called when angular1 destroys directive
Repro: http://plnkr.co/edit/s0FLVw1fIbG2mat954hu?p=preview

OnInit is called, but OnDestroy is not called when angular1 destroys an angular2 component via `ng-if`

### PR DESCRIPTION
**Pull request by andreialecu**
Fixes #6401 

Not sure why this is bound to `$remove`, the proper event is `$destroy` according to https://docs.angularjs.org/api/ng/function/angular.element

### CODE REVIEW

**Comment by tbosch** [comment]
Hi, thanks!
Could you create a test for this?

**Comment by andreialecu** [comment]
@tbosch: added test.

**Comment by texel** [comment]
This needs to be merged- without it, any angular1 app that uses angular2 components will never clean up those components.

**Comment by andreialecu** [comment]
@tbosch: mind giving this another look?

**Comment by laco0416** [comment]
@tbosch Please review. We need this fix seriously.

**Comment by andreialecu** [comment]
9 betas and 2 months later still no feedback here. 

Is there anything I need to change for it to get merged faster? It's absolutely critical for a few apps I've been working on.

angular1->2 upgraded apps simply can't function right without this fix - anything that uses an angular1 `ng-if` leaks resources like crazy. 

And it's not just resources, for example I have some logic that shows a modal when a certain websocket event arrives, and there is no way to unsubscribe from it when the user navigates away. The `ngOnDestroy` event is never fired in the angular2 component.

Paging more people (sorry): @tbosch @vsavkin @IgorMinar

File: modules/angular2/src/upgrade/downgrade_ng2_adapter.ts [hunk 1]
```diff
@@ -160,7 +160,7 @@ export class DowngradeNg2ComponentAdapter {
   }
 
   registerCleanup() {
-    this.element.bind('$remove', () => this.viewManager.destroyRootHostView(this.hostViewRef));
+    this.element.bind('$destroy', () => this.viewManager.destroyRootHostView(this.hostViewRef));
```
Comment (gkalpak) [inline modules/angular2/src/upgrade/downgrade_ng2_adapter.ts]: Nitpick: `.on()` is supposed to be preferred with `jQuery`.
(Without `jQuery`, `.bind()` is just an alias to `.on()`.)

**Comment by gkalpak** [comment]
FWIW, it LGTM :smiley:

**Comment by DrewynV** [comment]
__________________________________

Hi! We just wanted to let you know we receievd your email. Thank you for submitting your request. Our aim is to get back to you within 24 hours at the latest. If you have any additional information to add to your request, please reply to this email.


Also please visit our knowledgebase (http://www.support.magicdust.com.au) to browse for further info, we are adding to it all the time.

Or here is our industry report about the 11 Essentials for a Successful Website (http://www.magicdust.com.au/web-design/.

We look forward to helping you succeed online.

Thanks

The Team @ Magicdust Digital

http://support.magicdust.com.au/

Ph (02) 9979 7474
M-F | 9 am-5:00 pm AEST (AEDT in Summer) excluding Public Holidays

------------------------------------------------------------------------------------------------------------
This message was sent to  reply@reply.github.com in reference to Case #166503.
------------------------------------------------------------------------------------------------------------


[[9b3a0f269d81e49df9ef18d44104c57c59ef76a9-673073908]]

**Comment by IgorMinar** [comment]
@andreialecu sorry about the delay.. this should be out with the next beta.

**Comment by alxhub** [comment]
@andreialecu This is merged now via #7935 (I squashed your commits into one). Thanks!

**Pull request closed without merging**